src/main/java/com/massivecraft/factions/util/FlightUtil.java

```diff
@@ -54,32 +54,58 @@ public void run() {
             for (Player player : Bukkit.getOnlinePlayers()) {
                 FPlayer pilot = FPlayers.getInstance().getByPlayer(player);
                 if (pilot.isFlying() && !pilot.isAdminBypassing()) {
-                    if (enemiesNearby(pilot, 5)) {
-                        pilot.msg(TL.COMMAND_FLY_ENEMY_DISABLE);
-                        pilot.setFlying(false);
-                        if (pilot.isAutoFlying()) {
-                            pilot.setAutoFlying(false);
+                    findAndDisable(pilot, 5);
+                }
+            }
+        }
+
+        private void findAndDisable(FPlayer target, int radius) {
+            final List<Entity> nearbyEntities = target.getPlayer().getNearbyEntities(radius, radius, radius);
+            for (Entity entity : nearbyEntities) {
+                if (entity instanceof Player) {
+                    final FPlayer playerNearby = FPlayers.getInstance().getByPlayer((Player) entity);
+                    if (playerNearby.isAdminBypassing()) {
+                        continue;
+                    }
+                    if (playerNearby.getRelationTo(target) == Relation.ENEMY) {
+                        if (target.isFlying()) {
+                            disableFlight(target);
+                        }
+
+                        if (playerNearby.isFlying()) {
+                            disableFlight(playerNearby);
                         }
                     }
                 }
             }
         }
 
         public boolean enemiesNearby(FPlayer target, int radius) {
-            List<Entity> nearbyEntities = target.getPlayer().getNearbyEntities(radius, radius, radius);
+            final List<Entity> nearbyEntities = target.getPlayer().getNearbyEntities(radius, radius, radius);
             for (Entity entity : nearbyEntities) {
                 if (entity instanceof Player) {
-                    FPlayer playerNearby = FPlayers.getInstance().getByPlayer((Player) entity);
+                    final FPlayer playerNearby = FPlayers.getInstance().getByPlayer((Player) entity);
                     if (playerNearby.isAdminBypassing()) {
                         continue;
                     }
                     if (playerNearby.getRelationTo(target) == Relation.ENEMY) {
+                        if (playerNearby.isFlying()) {
+                            disableFlight(playerNearby);
+                        }
                         return true;
                     }
                 }
             }
             return false;
         }
+
+        private void disableFlight(FPlayer target) {
```

Comment:
Isn't there already a method for this in FPlayer

Reply:
There is no currently no method to disable flight in`FPlayer#` as you have to call `FPlayer#setFlying` and `FPlayer#setAutoFlying` seperately.

Reply:
Oh, looked more into this. The autofly setting is for enabling them to fly once they reach a place they can. Seems reasonable to *not* disable autofly here.